packages/components/src/components/post-linkarea/post-linkarea.tsx

```diff
@@ -1,33 +1,53 @@
-import { Component, Element, h, Host } from '@stencil/core';
+import { Component, Element, h, Host, State } from '@stencil/core';
 import { version } from '@root/package.json';
 
 const INTERACTIVE_ELEMENTS = ['a'].join(',');
 const INTERACTIVE_ELEMENTS_SELECTOR = `:where(${INTERACTIVE_ELEMENTS})`;
 
-type InteractiveElement = HTMLAnchorElement;
-
 @Component({
   tag: 'post-linkarea',
   styleUrl: 'post-linkarea.scss',
   shadow: true,
 })
 export class PostLinkarea {
+  private mutationObserver = new MutationObserver(this.checkInteractiveElements.bind(this));
+
   @Element() host: HTMLPostLinkareaElement;
 
+  @State() interactiveElements: NodeListOf<HTMLAnchorElement>;
+
   private dispatchClick({ ctrlKey, shiftKey, altKey, metaKey }: MouseEvent) {
-    const interactiveElement: InteractiveElement =
-      this.host.querySelector(`[data-link]${INTERACTIVE_ELEMENTS_SELECTOR}`) ??
-      this.host.querySelector(INTERACTIVE_ELEMENTS_SELECTOR);
+    this.interactiveElements[0].dispatchEvent(
```

Comment:
If you remove the error you'll need this:
```suggestion
    this.interactiveElements[0]?.dispatchEvent(
```